src/js/components/stock-movement-wizard/outbound/AddItemsPage.jsx

```diff
@@ -996,9 +996,13 @@ class AddItemsPage extends Component {
           return Promise.reject(new Error(this.props.translate('react.stockMovement.error.saveRequisitionItems.label', 'Could not save requisition items')));
         });
 
-      this.requestsQueue.enqueueRequest(
-        saveItemsRequest(payload),
-      );
+      if (this.props.isAutosaveEnabled) {
+        this.requestsQueue.enqueueRequest(
+          saveItemsRequest(payload),
+        );
+      } else {
+        await saveItemsRequest(payload)();
+      }
```

Comment:
lgtm but this currying looks weird to me, that it is required here `(payload)()` vs above `(payload)` - is it because of the implementation of the `TaskQueue` that it expects to get a function that it further runs, hence we need one more arrow function?